crates/sui-e2e-tests/tests/per_epoch_config_stress_tests.rs

```diff
@@ -63,21 +69,46 @@ async fn run_thread<F, Fut>(
 {
     let mut num_tx_succeeded = 0;
     let mut num_tx_failed = 0;
+    let mut epoch_map: BTreeMap<u64, Vec<_>> = BTreeMap::new();
     loop {
         let gas = test_env.get_latest_object_ref(&gas_id).await;
         let tx_data = tx_creation_func(test_env.clone(), gas).await;
         let tx = test_env.test_cluster.sign_transaction(&tx_data);
-        let Ok(effects) = test_env
-            .test_cluster
-            .wallet
-            .execute_transaction_may_fail(tx)
-            .await
-            .map(|r| r.effects.unwrap())
+        let Ok((effects, raw_effects)) =
+            execute_transaction_may_fail_with_raw_effects(&test_env.test_cluster.wallet, tx)
+                .await
+                .map(|r| (r.effects.unwrap(), r.raw_effects))
         else {
             // When epochs are short, it is possible that some transactions
             // keep getting sent at epoch boundaries and timeout eventually.
             continue;
         };
+
+        // make sure we have the raw effects
+        assert!(!raw_effects.is_empty());
+
+        let Ok(TransactionEffects::V2(v2_effects)) =
+            bcs::from_bytes::<TransactionEffects>(&raw_effects)
+        else {
+            panic!("Failed to deserialize raw effects into effects v2");
```

Comment:
`.expect`

Reply:
I was trying to collapse it all to one error (incorrect effects version, and invalid deser). Adding the `expect` might make it more complicated?